starfish/spots/_detector/spot_detector_results.py

```diff
@@ -0,0 +1,11 @@
+from typing import List, NamedTuple, Optional, Union
```

Comment:
I think it would be appropriate to refactor the `ConnectedComponentDecodingResult` from `combine_adjacent_features.py` to also live in this file.